main.tf

```diff
@@ -18,14 +22,14 @@ module "az_al_ad_application" {
 }
 
 resource "azurerm_resource_group" "lacework" {
-  name     = "${var.prefix}-group"
+  name     = "${var.prefix}-group-${random_id.uniq.hex}"
   location = var.location
 }
 
 # NOTE: storage name can only consist of lowercase letters and numbers,
 # and must be between 3 and 24 characters long
 resource "azurerm_storage_account" "lacework" {
-  name                      = "${var.prefix}storage"
+  name                      = "${var.prefix}storage${random_id.uniq.hex}"
```

Comment:
There was a limitation here where we only allow letters and numbers, do the `random_id` will only do those characters?

Reply:
The `hex` attribute should always return 0-9 or A-F - would it make sense to use `substr()` to cap at 24 characters? The default string is already 23 ('laceworkstorage' with 8 characters of random hex) so if someone does a custom prefix maybe we should trim the value automatically.